app/bolt_listeners.py

```diff
@@ -66,121 +62,156 @@ def respond_to_app_mention(
     client: WebClient,
     logger: logging.Logger,
 ):
-    if payload.get("thread_ts") is not None:
-        parent_message = find_parent_message(
-            client, context.channel_id, payload.get("thread_ts")
-        )
-        if parent_message is not None:
-            if is_no_mention_thread(context, parent_message):
-                # The message event handler will reply to this
-                return
-
-    wip_reply = None
-    # Replace placeholder for Slack user ID in the system prompt
-    system_text = build_system_text(SYSTEM_TEXT, TRANSLATE_MARKDOWN, context)
-    messages = [{"role": "system", "content": system_text}]
-
-    openai_api_key = context.get("OPENAI_API_KEY")
-    try:
-        if openai_api_key is None:
-            client.chat_postMessage(
+    logger.debug('respond to app mention')
+    if not FEATURE_RESPOND_VIA_MENTION and payload.get('channel_type') != "im":
+        return client.chat_postMessage(
                 channel=context.channel_id,
-                text="To use this app, please configure your OpenAI API key first",
+                thread_ts=payload["ts"],
+                text=translate(
+                    openai_api_key=context.get("OPENAI_API_KEY"),
+                    context=context,
+                    text=NO_APP_MENTION_MESSAGE,
+                ),
             )
-            return
+    
+    elif payload.get('channel_type') == "im":
+        return respond_in_dm(context=context, payload=payload, client=client, logger=logger)
+    
```

Comment:
This ensures that the bot tells the user it's unable to respond via an app mention.